src/renderer/components/layout/PrimaryLayout/Wallets/WalletMenu.tsx

```diff
@@ -72,72 +77,52 @@ const WalletMenu = ({ children, chains, wallets }: PropsWithChildren<Props>) =>
     }
   };
 
-  const handleCompleteVaultWalletCreation = () => {
-    toggleVaultModal();
-    setTimeout(() => navigate(Paths.ASSETS), DEFAULT_TRANSITION);
-  };
-
-  const handleCompleteWatchOnlyWalletCreation = () => {
-    toggleWatchOnlyModal();
-    setTimeout(() => navigate(Paths.ASSETS), DEFAULT_TRANSITION);
-  };
-
   return (
-    <>
-      <Popover className="relative">
-        <Popover.Button className="border border-container-border bg-left-navigation-menu-background rounded-md w-full shadow-card-shadow">
-          {children}
-        </Popover.Button>
-        <Transition
-          as={Fragment}
-          enter="transition ease-out duration-200"
-          enterFrom="opacity-0 translate-y-1"
-          enterTo="opacity-100 translate-y-0"
-          leave="transition ease-in duration-150"
-          leaveFrom="opacity-100 translate-y-0"
-          leaveTo="opacity-0 translate-y-1"
-        >
-          <Popover.Panel className="absolute z-40 rounded-md bg-token-container-background border border-token-container-border shadow-card-shadow mt-2">
-            {({ close }) => (
-              <section className={cn('relative w-[289px] bg-card-background')}>
-                <header className="px-5 py-3 flex items-center justify-between border-b border-divider">
-                  <SmallTitleText>{t('wallets.title')}</SmallTitleText>
-                  <DropdownButton
-                    options={dropdownOptions}
-                    className="w-[134px] py-2 h-8.5"
-                    title={t('wallets.addButtonTitle')}
-                  />
-                </header>
-
-                <div className="p-2 border-b border-divider">
-                  <SearchInput value={query} placeholder={t('wallets.searchPlaceholder')} onChange={setQuery} />
-                </div>
-
-                <ul className="flex flex-col divide-y divide-divider overflow-y-auto max-h-[530px]">
-                  {groupedWallets &&
-                    Object.entries(groupedWallets).map(([type, wallets]) => (
-                      <WalletGroup
-                        key={type}
-                        type={type as WalletType}
-                        wallets={wallets}
-                        onWalletClick={(wallet) => changeActiveAccount(wallet, close)}
-                      />
-                    ))}
-                </ul>
-              </section>
-            )}
-          </Popover.Panel>
-        </Transition>
-      </Popover>
-
-      <WatchOnly
```

Comment:
These 3 modals were executing all the time without render part but with useEffects and creating functions